docs/security.md

```diff
@@ -0,0 +1,121 @@
+# Security
+
+The OpenTelemetry Collector defaults to operating in a secure manner, but is
+configuration driven. This document captures important security aspects and
+considerations for the Collector.
```

Comment:
Who's the target audience? End-users, or component developers?

Reply:
Good point -- covering both to some extent makes sense. The initial version was more for end-users.